File: lib/bodies/section.dart
import 'package:mango_ui/keys.dart';

class Section {
  final String heading;
  final String text;
  final String imageUrl;
  final Key imageKey;

  Section(this.heading, this.text, this.imageUrl, this.imageKey);

  Map<String, dynamic> toJson() {
    return {
      "Heading": heading,
      "Text": text,
      "ImageUrl": imageUrl,
      "ImageKey": imageKey
    };
  }
}

File: core/section.go
package core

import (
	"github.com/louisevanderlith/husk/keys"
)

type Section struct {
	Heading  string  `hsk:"size(50)"`
	Text     string  `hsk:"size(512)"`
	ImageUrl string  `hsk:"null"`
	ImageKey *keys.TimeKey `hsk:"null"`
}
